src/librustc/ty/layout.rs

```diff
@@ -1713,13 +1713,13 @@ impl<'a, 'tcx> LayoutCx<'tcx, TyCtxt<'a, 'tcx, 'tcx>> {
                 let mut abi = Abi::Aggregate { sized: true };
                 if tag.value.size(dl) == size {
                     abi = Abi::Scalar(tag.clone());
-                } else if size > dl.pointer_size {
+                } else if !tag.is_bool() {
                     // HACK(nox): Blindly using ScalarPair for all tagged enums
-                    // that could lead to Option<u8> being handled as {i1, i8},
+                    // that could leads to Option<u8> being handled as {i1, i8},
```

Comment:
Did you mean to also remove "that could"?

Reply:
No. "Blindly using ScalarPair for all tagged enums that could [use ScalarPair] leads …"

Feel free to suggest me a better phrasing. :)

Reply:
Ah, I would've placed a comma there to disambiguate but I'm not sure that's valid grammar. Probably better to go with "where (it's) applicable" instead of "that could".